tests/adapters/test_prologix.py

```diff
@@ -23,45 +23,76 @@
 #
 
 import pytest
-import serial
 
 from pymeasure.adapters import PrologixAdapter
+from pymeasure.test import expected_protocol
 
-prefix = "\n".join(["++auto 0", "++eoi 1", "++eos 2", ""])
 
+def test_init():
+    with expected_protocol(
+            PrologixAdapter,
+            [("++auto 0", None), ("++eoi 1", None), ("++eos 2", None)]
+    ):
+        pass
 
-@pytest.fixture()
-def adapter():
-    return PrologixAdapter(serial.serial_for_url("loop://", timeout=0.2))
 
+def test_write():
+    with expected_protocol(
+            PrologixAdapter,
+            [("++auto 0", None), ("++eoi 1", None), ("++eos 2", None),
+             ("something", None)],
+    ) as adapter:
+        adapter.write("something")
 
-@pytest.mark.parametrize("test_input,expected", [("OUTP", prefix + "OUTP\n"),
-                                                 ("POWER 22 dBm", prefix + "POWER 22 dBm\n")])
-def test_adapter_write(test_input, expected, adapter):
-    adapter.write(test_input)
-    assert adapter.connection.read(len(expected) + 10) == expected.encode()
 
+def test_write_address():
```

Comment:
oh man it's great that we can test like this now!

Reply:
That's the reason, why the VISAAdapter init checks, whether the resource string is the ProtocolAdapter.